src/secp256k1.ts

```diff
@@ -15,10 +15,12 @@ function deriveScalar(bytes, discrim?: number) {
     }
     hasher.addU32(i)
     const key = hasher.first256BN()
+    /* istanbul ignore else */
     if (key.cmpn(0) > 0 && key.cmp(order) < 0) {
       return key
     }
   }
+  /* istanbul ignore next */
   throw new Error('impossible unicorn ;)')
```

Comment:
Why is this impossible to reach? Is it worth adding a code comment above here?

Reply:
This too originates from the initial import, could be something to audit -- and this code might be worth refactoring to remove this impossible case (assuming we understand it well enough)

Reply:
It's practically impossible. The order of the curve describes the (finite) amount of points on the curve. See: https://github.com/indutny/elliptic/blob/master/lib/elliptic/curves.js#L182

How often is an (essentially) random number (generated by sha512 algorithm) going to be larger than that number?

There's 2**32 aka "4 Billion and change" chances (the loop .... ) to get a number smaller than the order.  It's very rare that you will ever get past the first loop iteration. 

Do note that with TypeScript strict: true you would actually need the throw otherwise the function would return `BN | undefined`

disclaimer: I'm by *no* means an ECC expert however was the original author of this code.